packages/terra-filter-view/src/_Frame.jsx

```diff
@@ -0,0 +1,540 @@
+  dropdownAttrs: undefined,
+  isInvalid: false,
+  noResultContent: undefined,
+  onDeselect: undefined,
+  onSearch: undefined,
+  onSelect: undefined,
+  optionFilter: undefined,
+  placeholder: undefined,
+  value: undefined,
+  variant: Variants.DROPDOWN,
+};
+
+/* This rule can be removed when eslint-plugin-jsx-a11y is updated to ~> 6.0.0 */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+class Frame extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      excludeBlur: false,
+      isOpen: props.variant === Variants.LIST && props.showResultsInitially,
+      isFocused: false,
+      isPositioned: false,
+      searchValue: '',
+      showResults: props.showResultsInitially,
+    };
+
+    this.setInput = this.setInput.bind(this);
+    this.getDisplay = this.getDisplay.bind(this);
+    this.clearSearch = this.clearSearch.bind(this);
+    this.clearSearchTimeout = this.clearSearchTimeout.bind(this);
+    this.delaySearch = this.delaySearch.bind(this);
+    this.openDropdown = this.openDropdown.bind(this);
+    this.closeDropdown = this.closeDropdown.bind(this);
+    this.toggleDropdown = this.toggleDropdown.bind(this);
+    this.positionDropdown = this.positionDropdown.bind(this);
+    this.handleBlur = this.handleBlur.bind(this);
+    this.handleFocus = this.handleFocus.bind(this);
+    this.handleMouseEnter = this.handleMouseEnter.bind(this);
+    this.handleMouseLeave = this.handleMouseLeave.bind(this);
+    this.handleSelect = this.handleSelect.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleMouseDown = this.handleMouseDown.bind(this);
+    this.handleInputMouseDown = this.handleInputMouseDown.bind(this);
+  }
+
+  componentDidUpdate(previousProps, previousState) {
+    if (this.props.variant === Variants.DROPDOWN && Util.shouldPositionDropdown(previousState, this.state, this.dropdown)) {
+      clearTimeout(this.debounceTimer);
+      this.debounceTimer = setTimeout(this.positionDropdown, !previousState.isOpen ? 0 : 100);
+    }
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.debounceTimer);
+  }
+
+  setInput(input) {
+    this.input = input;
+  }
+
+  getDisplay() {
+    const { searchValue } = this.state;
+    const { disabled, placeholder } = this.props;
+
+    const inputAttrs = {
+      disabled,
+      placeholder,
+      ref: this.setInput,
+      onChange: this.handleSearch,
+      onMouseDown: this.handleInputMouseDown,
+      'aria-label': 'search',
+      className: cx('search-input', { 'right-padding': this.state.isFocused && this.state.searchValue }),
+    };
+
+    return <input {...inputAttrs} value={searchValue} />;
+  }
+
+  /**
+   * Clears the search-bar content.
+   */
+  clearSearch() {
+    if (this.state.searchValue !== '') {
+      this.setState({ searchValue: '', excludeBlur: false });
+    }
+    // Refocus input after clearing
+    this.input.focus();
```

Comment:
Blur and focus is a dangerous game of catch. To prevent the blur entirely you may be able to prevent default on the event. Examples of this can be found throughout the Select component and this component. 

https://github.com/cerner/terra-core/blob/9bade521c11c07ca67263b439936e8fed4a742ab/packages/terra-form-select/src/_Dropdown.jsx#L58